pkg/cli/oicd_connect.go

```diff
@@ -0,0 +1,71 @@
+package cli
+
+import (
+	"net"
+	"net/http"
+	"time"
+
+	"github.com/projectcontour/contour-authserver/pkg/auth"
+	"github.com/projectcontour/contour-authserver/pkg/config"
+	"github.com/spf13/cobra"
+
+	"github.com/allegro/bigcache"
+
+	ctrl "sigs.k8s.io/controller-runtime"
+)
+
+//NewOIDCConnect - start server as OIDC and take in 'config' file as parameter...
+func NewOIDCConnect() *cobra.Command {
+	cmd := cobra.Command{
+		Use:   "oidc Server [OPTIONS]",
+		Short: "Run a OIDC authentication server",
+		Args:  cobra.NoArgs,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			log := ctrl.Log.WithName("auth.oidcConnect")
+
+			cfgFile, err := cmd.Flags().GetString("config")
+			cfg, err := config.NewConfig(cfgFile)
+
+			if err != nil {
+				return ExitError{EX_CONFIG, err}
+			}
+
+			if cfg == nil {
```

Comment:
This should be an error since a nil will make it panic later. Even better wold be for the config to be guaranteer non-nil when there is no error.

Reply:
I think that in the version I'm reviewing it's not possible for this to be nil here? It will always have at least the defaults? If so, this check will never be used.